src/protocols/secure_channel/CASESession.cpp

```diff
@@ -974,140 +1074,143 @@ CHIP_ERROR CASESession::TryResumeSession(SessionResumptionStorage::ConstResumpti
+        ChipLogByteSpan(SecureChannel, parsedSigma1.destinationId);
 
-    // ParseSigma1 ensures that:
-    // mRemotePubKey.Length() == initiatorPubKey.size() == kP256_PublicKey_Length.
-    memcpy(mRemotePubKey.Bytes(), initiatorPubKey.data(), mRemotePubKey.Length());
+        // FindLocalNodeFromDestinationId returns CHIP_ERROR_KEY_NOT_FOUND if validation of DestinationID fails, which will trigger
+        // status Report with ProtocolCode = NoSharedTrustRoots.
+        mNextStep = Step::kSendStatusReport;
 
-    MATTER_LOG_METRIC_BEGIN(kMetricDeviceCASESessionSigma2);
-    err = SendSigma2();
-    if (CHIP_NO_ERROR != err)
-    {
-        MATTER_LOG_METRIC_END(kMetricDeviceCASESessionSigma2, err);
+        return err;
     }
-    SuccessOrExit(err);
-
-    mDelegate->OnSessionEstablishmentStarted();
-
-exit:
 
-    if (err == CHIP_ERROR_KEY_NOT_FOUND)
-    {
-        SendStatusReport(mExchangeCtxt, kProtocolCodeNoSharedRoot);
-        mState = State::kInitialized;
-    }
-    else if (err != CHIP_NO_ERROR)
-    {
-        SendStatusReport(mExchangeCtxt, kProtocolCodeInvalidParam);
-        mState = State::kInitialized;
-    }
-    return err;
+    return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR CASESession::SendSigma2Resume()
+CHIP_ERROR CASESession::PrepareSigma2Resume(EncodeSigma2ResInputs & output)
 {
-    MATTER_TRACE_SCOPE("SendSigma2Resume", "CASESession");
-    size_t max_sigma2_resume_data_len =
-        TLV::EstimateStructOverhead(SessionResumptionStorage::kResumptionIdSize, CHIP_CRYPTO_AEAD_MIC_LENGTH_BYTES,
-                                    sizeof(uint16_t), SessionParameters::kEstimatedTLVSize);
-
-    System::PacketBufferTLVWriter tlvWriter;
-    System::PacketBufferHandle msg_R2_resume;
-    TLV::TLVType outerContainerType = TLV::kTLVType_NotSpecified;
+    MATTER_TRACE_SCOPE("PrepareSigma2Resume", "CASESession");
 
     // Validate that we have a session ID allocated.
     VerifyOrReturnError(GetLocalSessionId().HasValue(), CHIP_ERROR_INCORRECT_STATE);
+    output.responderSessionId = GetLocalSessionId().Value();
+
+    // Generate a new resumption ID
+    ReturnErrorOnFailure(DRBG_get_bytes(mNewResumptionId.data(), mNewResumptionId.size()));
+    output.resumptionId = mNewResumptionId;
+
+    ReturnErrorOnFailure(GenerateSigmaResumeMIC(ByteSpan(mInitiatorRandom), mNewResumptionId, ByteSpan(kKDFS2RKeyInfo),
+                                                ByteSpan(kResume2MIC_Nonce), output.resumeMICSpan));
+
+    VerifyOrReturnError(mLocalMRPConfig.HasValue(), CHIP_ERROR_INCORRECT_STATE);
+    output.responderMrpConfig = &mLocalMRPConfig.Value();
+
+    mState = State::kSentSigma2Resume;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CASESession::EncodeSigma2Resume(System::PacketBufferHandle & msg_R2_resume, EncodeSigma2ResInputs & input)
+{
+    MATTER_TRACE_SCOPE("EncodeSigma2Resume", "CASESession");
+    size_t max_sigma2_resume_data_len = TLV::EstimateStructOverhead(SessionResumptionStorage::kResumptionIdSize, // resumptionID
```

Comment:
suggest being consistent on naming. We have camelCase and snake_case mixed here.

Reply:
I have changed all instances of snake_case to camelCase in the Methods I touched